src/lighthouse.rs

```diff
@@ -163,30 +185,49 @@ fn quorum_compute(
         }
     }
 
+    // Minimum quorum size check.
     if healthy_participants.len() < opt.min_replicas as usize {
         return (
             None,
             format!(
-                "No quorum, only have {} participants, need {} {}",
+                "No quorum, only have {} participants, need min_replicas {} {}",
                 healthy_participants.len(),
                 opt.min_replicas,
                 metadata
             ),
         );
     }
 
+    // Avoid split brain by requiring at least half of the known alive workers.
+    if healthy_participants.len() <= healthy_replicas.len() / 2 {
```

Comment:
Would you not get split brain if you have equal sizes of healthy workers? E.g. healthy_participants= 5, healthy_replicas= 10.

Reply:
This check blocks the quorum so for the positive case we'd end up with

```rs
if 5 <= 10 / 2 {
    return (None, ...)
}
```

so we would make a quorum of minimum size 6